### PR TITLE
Better flash effects

### LINKED ISSUE #1539: M84 Flashbang has low effect radius
ACE3 Version: 3.0.1
(indicate if stable or dev, if dev indicate the commit the version is based on)

**Mods:**
* @cba_a3
* @ace3

**Placed ACE3 Modules:**
* *None*

**Description:**
* *M84 continues to fly after explosion*
* *As you can see on video below M84 seems to have a bit too low effect radius, it should still blind me by staring at it like that I think.*

https://www.youtube.com/watch?v=ZLHWfz9rp2s&feature=youtu.be

**Steps to reproduce:**
* *Throw M84 in the air and observe.*

**Where did the issue occur?**
*"Multiplayer", "Singleplayer"*

**RPT log file:**
*Not needed I think.*

### PR DESCRIPTION
**Pull request by esteldunedain**
### When merged this pull request will:

1. Make sure the light effect is rendered even if the grenade explodes further than 20m from the player. This provides much better feedback that the grenade exploded, even if the player is not affected.
2. Increase the amount of hearing damage from the bang. Should work nicely with the new hearing overhaul.
3. Increase the maximum range for blinding effect to 25m, but with a higher-than-linear falloff. Close #1539.
4. Make units heavily affected by the blast turn away from the light, as per @bux578 request.

Bottom line, this should improve both the UX of throwing flashbangs and their effectiveness.

### CODE REVIEW

**Comment by bux578** [comment]
:+1: 

How strong will AI be affected by this?
I think AI could need some more love concerning flashbangs

**Comment by esteldunedain** [comment]
> How strong will AI be affected by this?

I didn't change anything for AI, except for rotating them away. What are you thinking should be done?

**Comment by bux578** [comment]
I just tested it. The rotating away is very welcome. Good job.

**Comment by bux578** [comment]
https://youtu.be/6q5YF_nyZSc

**Comment by esteldunedain** [comment]
Nice:

1- the last one shot you because he was too far from the grenade. Maybe we should increase the range another 5m?

2- The flash of light is visible through the wall due to arma awfull illumination technology. Hopefully if they ever change to a deferred technique that should be fixed.

**Comment by bux578** [comment]
Question: The rotation is 180° from current direction? Or 180° from the direction to the flashbang?

If it is from current direction, AI units that look through a window and have the door behind them would rotate towards the door.

**Comment by esteldunedain** [comment]
The rotation is 180° away from the grenade

**Comment by bux578** [comment]
> Maybe we should increase the range another 5m

I'll ask my BW aquaintance.

**Comment by jokoho48** [comment]
@esteldunedain can you add A Animation for that like in the mod: http://www.armaholic.com/page.php?id=21139

this is a Vanilla Animation if i remeber right

**Comment by esteldunedain** [comment]
> @esteldunedain can you add A Animation for that like in the mod: http://www.armaholic.com/page.php?id=21139

Looks cool, but I don't what animation is that. @CorruptedHeart, are you the author?

**Comment by bux578** [comment]
A M84 should totally cover this room.

**Comment by esteldunedain** [comment]
> A M84 should totally cover this room.

It should cover it now

**Comment by bux578** [comment]
:+1: merge merge merge :smile:
